transaction/core/src/signer/traits.rs

```diff
@@ -1,8 +1,62 @@
-use crate::ring_signature::{CryptoRngCore, MLSAGError, RingMLSAG, Scalar, SignableInputRing};
-use alloc::string::String;
+use crate::{
+    ring_signature::{CryptoRngCore, MLSAGError, ReducedTxOut, RingMLSAG, Scalar},
+    Amount,
+};
+use alloc::{string::String, vec::Vec};
 use displaydoc::Display;
-use mc_crypto_keys::KeyError;
+use mc_crypto_keys::{KeyError, RistrettoPrivate};
 use serde::{Deserialize, Serialize};
+use zeroize::Zeroize;
+
+/// A representation of the part of the input ring needed to create an MLSAG
+#[derive(Clone, Debug)]
+pub struct SignableInputRing {
+    /// A reduced representation of the TxOut's in the ring. For each ring
+    /// member we have only:
+    /// * The onetime-address (tx_out.target_key)
+    /// * The compressed commitment (tx_out.amount.commitment)
+    pub members: Vec<ReducedTxOut>,
```

Comment:
Out of curiosity, what do we actually get from passing around the `ReducedTxOut` and not just `TxOut`? Is there a flow where the `TxOut` is not available to us?

Reply:
I think it's mainly that it will be easier to split the ring signature stuff out of the transaction-core crate -- initially I think this stuff was in it's own crate below the blockchain stuff, and so it couldn't use the `TxOut` type directly

Reply:
I guess I don't think we should we need an MCIP because this isn't a protocol rules change and it isn't creating new network interfaces -- I think it's just a software engineering project

There have been a lot of design discussions in the hardware wallets project group -- this is also just an experiment right now and we aren't set on this as a direction. If you want to be more involved we can invite you to our meetings